src/gluonts/distribution/gamma.py

```diff
@@ -0,0 +1,144 @@
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+import math
+from functools import partial
+from typing import Dict, Optional, Tuple, List
+
+# Third-party imports
+import numpy as np
+
+# First-party imports
+from gluonts.model.common import Tensor
+from gluonts.support.util import erf, erfinv
+from gluonts.core.component import validated
+
+# Relative imports
+from .distribution import Distribution, _sample_multiple, getF, softplus
+from .distribution_output import DistributionOutput
+
+
+class Gamma(Distribution):
+    r"""
+    Gamma distribution.
+
+    Parameters
+    ----------
+    alpha
+        Tensor containing the shape parameters, of shape `(*batch_shape, *event_shape)`.
+    beta
+        Tensor containing the rate parameters, of shape `(*batch_shape, *event_shape)`.
+    F
+    """
+
+    is_reparameterizable = True
+
+    @validated()
+    def __init__(self, alpha: Tensor, beta: Tensor, F=None) -> None:
+        self.alpha = alpha
+        self.beta = beta
+        self.F = (
+            F if F else getF(alpha)
+        )  # assuming alpha and beta of same type
+
+    @property
+    def batch_shape(self) -> Tuple:
+        return self.alpha.shape
+
+    @property
+    def event_shape(self) -> Tuple:
+        return ()
+
+    @property
+    def event_dim(self) -> int:
+        return 0
+
+    def log_prob(self, x: Tensor) -> Tensor:
+        F = self.F
+        alpha, beta = self.alpha, self.beta
+
+        return (
+            alpha * F.log(beta)
+            - F.gammaln(alpha)
+            + (alpha - 1) * F.log(x)
+            - beta * x
+        )
+
+    @property
+    def mean(self) -> Tensor:
+        return self.alpha / self.beta
+
+    @property
+    def stddev(self) -> Tensor:
+        return self.F.sqrt(self.alpha) / self.beta
+
+    def sample(
+        self, num_samples: Optional[int] = None, dtype=np.float32
+    ) -> Tensor:
+        return _sample_multiple(
+            partial(self.F.sample_gamma, dtype=dtype),
+            alpha=self.alpha,
+            beta=1.0 / self.beta,  # because MXNet doesnt make sense
+            num_samples=num_samples,
+        )
+
+    def sample_rep(
+        self, num_samples: Optional[int] = None, dtype=np.float32
+    ) -> Tensor:
+        def s(alpha: Tensor, beta: Tensor) -> Tensor:
+            return self.F.sample_gamma(alpha=alpha, beta=beta, dtype=dtype)
+
+        return _sample_multiple(
+            s,
+            alpha=self.alpha,
+            beta=1.0 / self.beta,
+            num_samples=num_samples,  # because MXNet doesnt make sense
+        )
```

Comment:
I think you can remove this definition for now, since it's the same as `sample`. The idea of having `sample_rep` is to be able to differentiate through samples.

In the case of Gaussian, you can express samples from a generic Gaussian distribution by scaling and translating samples from a standard normal (see [here](https://github.com/awslabs/gluon-ts/blob/fbcb244da611767ecf961955a27ad0f39b029133/src/gluonts/distribution/gaussian.py#L103-L107)): this allows you e.g. to get the derivative of the samples wrt mu and sigma.

I'm not sure this can be done for Gamma (it can be done wrt the beta parameter, not sure about alpha).

Reply:
For large Alpha there is a convenient way to sample based on the Normal Distribution. 
In general however, sampling from a Gamma distribution is more involved(see: https://www.hongliangjie.com/2012/12/19/how-to-generate-gamma-random-variables/ or http://home.iitk.ac.in/~kundu/paper120.pdf) which entails loops and ifs.
So I could implement one of the methods, but it would be probably painfully inefficient in python.

So ok, I"ll remove sample_rep.

Reply:
Yeah, I think it's OK for now to just remove it and set `is_reparameterizable = False`